src/deepmol/loaders/loaders.py

```diff
@@ -154,68 +90,57 @@ def _get_dataset(dataset_path: str,
             path to the dataset file
         fields: List[str]
             fields to keep
-        sep: str
-            separator
-        header: int
-            the row where the header is
         chunk_size: int
             size of the shard to load
+        kwargs:
+            Keyword arguments to pass to pandas.read_csv.
 
         Returns
         -------
         pd.DataFrame
             Dataframe with chunk size.
         """
-        return load_csv_file(dataset_path, fields, sep, header, chunk_size)
+        return load_csv_file(dataset_path, fields, chunk_size, **kwargs)
 
-    def create_dataset(self, sep: str = ',', header: int = 0, in_memory: bool = True):
+    def create_dataset(self, **kwargs) -> NumpyDataset:
         """
         Creates a dataset from the CSV file.
 
         Parameters
         ----------
-        sep: str
-            separator
-        header: int
-            the row where the header is
-        in_memory: bool
-            whether to load the dataset in memory or not
+        kwargs:
+            Keyword arguments to pass to pandas.read_csv.
 
         Returns
         -------
         NumpyDataset
             Dataset with the data.
         """
-        if in_memory:
-            dataset = self._get_dataset(self.dataset_path, fields=self.fields2keep, sep=sep, header=header,
-                                        chunk_size=self.shard_size)
-
-            mols = dataset[self.mols_field].to_numpy()
-
-            if self.features_fields is not None:
-                X = dataset[self.features_fields].to_numpy()
-            else:
-                X = None
+        dataset = self._get_dataset(self.dataset_path, fields=self.fields2keep, chunk_size=self.shard_size, **kwargs)
 
-            if self.labels_fields is not None:
-                y = dataset[self.labels_fields].to_numpy()
-            else:
-                y = None
+        mols = dataset[self.mols_field].to_numpy()
 
-            if self.id_field is not None:
-                ids = dataset[self.id_field].to_numpy()
-            else:
-                ids = np.array([i for i in range(dataset.shape[0])])
+        if self.features_fields is not None:
+            X = dataset[self.features_fields].to_numpy()
+        else:
+            X = None
 
-            return NumpyDataset(mols=mols,
-                                X=X,
-                                y=y,
-                                ids=ids,
-                                features2keep=self.features2keep,
-                                n_tasks=self.n_tasks)
+        if self.labels_fields is not None:
+            y = dataset[self.labels_fields].to_numpy()
+        else:
+            y = None
 
+        if self.id_field is not None:
+            ids = dataset[self.id_field].to_numpy()
         else:
-            raise NotImplementedError
+            ids = np.array([i for i in range(dataset.shape[0])])
```

Comment:
Shouldn't this be consistent with the NumpyDataset implementation, where you use uuid4? I think you should either leave it as None and the NumpyDataset will take care of it or define it as in NumpyDataset